### ISSUE TITLE
Scheduled cache values only update themselves once 

### ISSUE THREAD

**Issue by kit-ty-kate** [ocurrent/ocurrent#180]
I'm not sure whether or not I built the test case right or if I missed something but this code:
```ocaml
let () = Random.self_init ()

module Rand = struct
  type t = No_context

  let id = "rand"

  module Key = struct
    type t = int

    let digest = string_of_int
  end

  module Value = struct
    type t = int

    let marshal = string_of_int
    let unmarshal = int_of_string
  end

  let build No_context job _key =
    let open Lwt.Infix in
    Current.Job.start ~level:Current.Level.Harmless job >>= fun () ->
    print_endline "build triggered";
    Lwt.return (Ok (Random.int 100))

  let pp = Fmt.int

  let auto_cancel = true
end

module Rand_cache = Current_cache.Make(Rand)

let schedule = Current_cache.Schedule.v ~valid_for:(Duration.of_sec 1) ()

let () =
  let v = Rand_cache.get ~schedule No_context 0 in
  Lwt_main.run @@
  Current.Engine.thread @@
  Current.Engine.create @@ fun () ->
  let open Current.Syntax in
  let+ _ : int =
    Current.component "test" |>
    let> () = Current.return () in
    Current_incr.of_cc @@
    Current_incr.read v @@ fun (v, job) ->
    begin match v with
    | Error (`Active `Ready) -> print_endline "ready"
    | Error (`Active `Running) -> print_endline "running"
    | Error (`Msg m) -> print_endline ("error: "^m)
    | Ok v -> print_endline (string_of_int v)
    end;
    Current_incr.write ~eq:(fun _ _ -> false) (v, job)
  in
  ()
```
doesn't end up being an infinit loop printing random numbers between 0 and 100 like I thought it would be, but only stopped after the second try:
```
$ dune exec ./test.exe
ready                
6
# waits 1 second
build triggered
27
# waits forever
^C
```
Have I missed something?

*side-note: I could've used the `Current.state` function instead of the low-level functions from Current_incr, I'm not sure why I did, but it does the same thing anyway.*

**Comment by kit-ty-kate** [comment]
Ah apparently if we add the following line to the `build` function it works:
```
Current.Process.exec ~cancellable:true ~job ("", [|"true"|]) >>= fun _ ->
```
Is that expected?

**Comment by talex5** [comment]
Fixed in #185 - thanks for the reproduction case!
